src/Service/Fido2ServiceEndpoints.cs

```diff
@@ -38,16 +38,15 @@ public class Fido2ServiceEndpoints : IFido2Service
         _eventLogger = eventLogger;
     }
 
-    private async Task Init()
+    private async Task Init(CancellationToken cancellationToken)
     {
-        await _tokenService.InitAsync();
+        await _tokenService.InitAsync(cancellationToken);
```

Comment:
> This feels to me like a bit of a footgun (to share the same cancellationToken).

Why? 🤔

Reply:
I may be wrong, but I've only ever seen cancellationToken being passed into the actual methods, not the "constructor" of the class.

We may of course pass cancellationToken to `InitAsync` to make sure we can abort anything we do in that method, but the way the code is implemented is that we set the cancellationToken as member of the class and it's utilized for all other methods of that class. That is a design pattern I haven't seen before and it can have benefits, but to me it seems the norm is to give the caller of a method the control to abort a single call. This is debatable, one could argue that it's a feature, but to me at least it's a new and surprising behaviour that would introduce severe bugs if the the lifetime of the class would be changed.

Reply:
Yeah, storing cancellation tokens as fields is not _too_ unusual either, but it's not done here.